superset/views/core.py

```diff
@@ -273,6 +274,20 @@ def pre_delete(self, obj):
     def _delete(self, pk):
         DeleteMixin._delete(self, pk)
 
+    def check_extra(self, db):
+        # this will check whether json.loads(extra) can succeed
+        try:
+            extra = db.get_extra()
+        except Exception as e:
+            raise Exception('Extra field cannot be decoded by JSON. {}'.format(str(e)))
+
+        # this will check whether 'metadata_params' is configured correctly
+        try:
+            MetaData(**extra.get('metadata_params', {}))
```

Comment:
A cool thing you can do here is actually check with arguments the `MetaData` constructor accepts, using the `inspect` standard library module:

```python
>>> from sqlalchemy import MetaData
>>> import inspect
>>> inspect.getargspec(MetaData)
ArgSpec(args=['self', 'bind', 'reflect', 'schema', 'quote_schema', 'naming_convention', 'info'], varargs=None, keywords=None, defaults=(None, False, None, None, immutabledict({'ix': 'ix_%(column_0_label)s'}), None))
```

This way you can check if any of the keys in `extra['metadata_params']` is not in the `args` of the `ArgSpec`, and print an error message saying "The key 'foo' is not valid!".

Note that even if we do this we should still keep the `try/except` block, since even if the keys are valid the values could be wrong.

Reply:
@youngyjd let me know if you're planning to do the `inspect` magic in this PR, or if I should go ahead and merge this as is.

Reply:
I will try to see how I can use `inspect`. Not having enough time to play around with it. Will update.